src/factories/nodeTaskRun.ts

```diff
@@ -18,33 +17,28 @@ export async function taskRunContainerFactory() {
   container.addChild(label)
 
   async function render(node: RunGraphNode): Promise<BoundsContainer> {
-    const label = await renderLabel(node)
-    const bar = await renderBar(node)
+    await Promise.all([
+      renderBar(node),
+      renderLabel(node.label),
+    ])
 
-    label.position = await getLabelPosition(label, bar)
+    updateLabel()
 
     return container
   }
 
-  async function getLabelPosition(label: BitmapText, bar: BoundsContainer): Promise<Pixels> {
-    const config = await waitForConfig()
-
-    // todo: this should probably be nodePadding
-    const margin = config.styles.nodePadding
-    const inside = bar.width > margin + label.width + margin
-    const y = bar.height / 2 - label.height / 2
-
-    if (inside) {
-      return {
-        x: margin,
-        y,
-      }
-    }
-
-    return {
-      x: bar.width + margin,
-      y,
-    }
+  function updateLabel(): void {
+    const colorOnNode = config.styles.colorMode === 'dark'
+      ? config.styles.textDefault
+      : config.styles.textInverse
+
+    const padding = config.styles.nodePadding
+    const inside = bar.width > padding + label.width + padding
+    const x = inside ? padding : bar.width + padding
+    const y = config.styles.nodeHeight / 2 - label.height / 2
```

Comment:
Tossed this in the PR: updated the `y` assignment so that it doesn't jump when the bar is selected